tests/test_websocket_integration.py

```
@@ -0,0 +1,128 @@
+            assert "get_tws_status" in agent_msg["metadata"]["tools_used"]
+            
+            # Verify trace ID propagation (correlation_id in payload)
+            assert "correlation_id" in agent_msg
+
+def test_websocket_auth_failure(app):
+    client = TestClient(app)
+    
+    # Mock authentication service to fail
+    with patch("resync.api.auth.service.AuthService.verify_token", return_value=False):
+        with pytest.raises(WebSocketDisconnect) as excinfo:
+            with client.websocket_connect("/ws/tws-general?token=invalid") as websocket:
```

Ruff (0.15.1)

[error] Local variable `websocket` is assigned to but never used

Remove assignment to unused variable `websocket`

(F841)